python/ray/serve/backend_worker.py

```diff
@@ -251,6 +253,31 @@ def get_runner_method(self, request_item: Query) -> Callable:
             return self.callable
         return getattr(self.callable, method_name)
 
+    async def ensure_serializable_response(self, response: Any) -> Any:
+        if isinstance(response, starlette.responses.StreamingResponse):
+            # response contains a generator/iterator which is not serializable.
+            # Exhaust the generator/iterator and store the results in a buffer.
+            body_buffer = []
+
+            async def mock_send(message):
+                if (message["type"] == "http.response.body"):
+                    body_buffer.append(message["body"])
+
+            async def mock_receive():
+                # This is called in a while loop in response(), so sleep a bit.
+                await asyncio.sleep(1)
+                return {"type": None}
+
+            await response(None, mock_receive, mock_send)
```

Comment:
the first argument is `scope` dictionary right? can we make it keyworded so at least we know what argument is ignored: `await response(scope=None, receive=..., ...)`